File: server/java/src/main/java/com/onlinepayments/example/api/models/additionalpaymentactions/AdditionalPaymentActionRequest.java
package com.onlinepayments.example.api.models.additionalpaymentactions;

import com.onlinepayments.example.service.domain.common.enums.Currency;

import java.math.BigDecimal;

public record AdditionalPaymentActionRequest(
        BigDecimal amount,
        Currency currency,
        Boolean isFinal
) {}

File: server/nodejs/src/presentation/models/payment/additional-payment-action/additional-payment-action-request.ts
import { FastifyRequest } from 'fastify';
import { Currency } from '../../../../business/domain/common/enums/currency';

export interface AdditionalPaymentActionRequest {
    amount?: number;
    currency?: Currency;
    isFinal?: boolean;
}

export const AdditionalPaymentActionRequest = {
    fromApiRequest(request: FastifyRequest): AdditionalPaymentActionRequest {
        const data = (request.body as any) || {};

        return {
            amount: data.amount,
            currency: data.currency,
            isFinal: data.isFinal,
        };
    },
};
